arangod/Graph/Graph.cpp

```diff
@@ -239,6 +234,37 @@ void Graph::insertOrphanCollections(VPackSlice const arr) {
   }
 }
 
+std::set<std::string> Graph::vertexCollectionsUsedAsSatellites(
+    TRI_vocbase_t& vocbase) const {
+  std::set<std::string> list{};
+
+  std::shared_ptr<LogicalCollection> logicalCollection;
+  for (auto const& vertexCollection : vertexCollections()) {
+    logicalCollection =
+        graph::GraphManager::getCollectionByName(vocbase, vertexCollection);
+    if (logicalCollection != nullptr && logicalCollection->isSatellite()) {
+      list.emplace(vertexCollection);
+    }
+  }
+
+  return list;
+}
+
+std::string Graph::calculateMd5() const {
```

Comment:
This is a pretty expensive operation, and requires every graph to be serialized twice.
Are we protected against races here?
(e.g. someone modifing the graph in between original  serialization and serialization for md5?)

As an alternative:
Is it possible to not inline the md5 sums, but return them next to the serialized result?
e.g.:
```
{
  graphs: [{graph1}, {graph2}],
  details: {
    global: "md5-of-all-graphs",
    graphs: ["md5-of-graph1", "md5-of-graph2"]
  }
}
```

This way we can calculate the MD5Sum based on the graphs array that we actually return, and are save against races. We are also more save against changing the content of the graph JSON in future versions, as the MD5 method simply takes any slice, and needs no knowledge about graph internals., like it somehow does now